lib/liquid/template_factory.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module Liquid
+  class TemplateFactory
+    def for(_template_name)
+      Liquid::Template.new
+    end
+  end
```

Comment:
Why bother having a class for this, since it is just a method without state.  Instead, this could just be

```ruby
  module TemplateFactory
    def self.for(_template_name)
      Liquid::Template.new
    end
  end
```

and use `Liquid::TemplateFactory` instead of `Liquid::TemplateFactory.new` as the default when `context.registers[:template_factory]` isn't set.

Reply:
I had gone with a class level factory method at first but since all other registries in the liquid `context` are objects, I decided to make it an instance method. This way, registers remain object and avoids surprise. I'll leave it as is if that's ok.

Reply:
> This way, registers remain objects

A singleton object is an object.  You also wouldn't need to memoize the default in the registry, since that was just an optimization to avoid object allocation.

Reply:
You are correct, I should have been more clear. I meant that all entries in the register get initialized `*.new` as opposed to being a class. I wanted to follow this pattern 🤷‍♂